Issue with BaseModelica export and records

Please consider this MWE:
```modelica
model TestFluid
  package Fluid = SOFCPoliMi.Media.MainClasses.SOS_CO2.SOS10ComponentsModelica;
  Fluid fluid(p_start = 1e5, T_start = 300);
equation
  fluid.p = 1e5;
  fluid.T = 300;
  fluid.X = 0.1*ones(10);
end TestFluid;
```
The model simulates without problems. If I export it with --baseModelica and then re-read it and try to simulate it, it fails at compile time with errors such as:
```
In file included from _omcQuot_2754657374466c75696427_records.c:5:
./_omcQuot_2754657374466c75696427_functions.h:82:1: error: unknown type name '_omcQ_27TestFluid_27__omcQ_27Modelica_5F_5FMedia_5F_5FIdealGases_5F_5FCommon_5F_5FDataRecord_27'; did you mean '_omcQ_27TestFluid_27__omcQ_27Modelica_2EMedia_2EIdealGases_2ECommon_2EDataRecord_27'?
   82 | _omcQ_27TestFluid_27__omcQ_27Modelica_5F_5FMedia_5F_5FIdealGases_5F_5FCommon_5F_5FDataRecord_27 omc__omcQ_27TestFluid_27__omcQ_27Modelica_5F_5FMedia_5F_5FIdealGases_5F_5FCommon_5F_5FDataRecord_27 (threadData_t *threadData, modelica_string omc__omcQ_27name_27, modelica_real omc__omcQ_27MM_27, modelica_real omc__omcQ_27Hf_27, modelica_real omc__omcQ_27H0_27, modelica_real omc__omcQ_27Tlimit_27, real_array omc__omcQ_27alow_27, real_array omc__omcQ_27blow_27, real_array omc__omcQ_27ahigh_27, real_array omc__omcQ_27bhigh_27, modelica_real omc__omcQ_27R_5Fs_27);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      | _omcQ_27TestFluid_27__omcQ_27Modelica_2EMedia_2EIdealGases_2ECommon_2EDataRecord_27
./_omcQuot_2754657374466c75696427_functions.h:25:3: note: '_omcQ_27TestFluid_27__omcQ_27Modelica_2EMedia_2EIdealGases_2ECommon_2EDataRecord_27' declared here
   25 | } _omcQ_27TestFluid_27__omcQ_27Modelica_2EMedia_2EIdealGases_2ECommon_2EDataRecord_27;
      |   ^
./_omcQuot_2754657374466c75696427_functions.h:91:155: error: unknown type name '_omcQ_27TestFluid_27__omcQ_27Modelica_5F_5FMedia_5F_5FIdealGases_5F_5FCommon_5F_5FDataRecord_27'
```
I guess there is something wrong in the way records are handled in the Base Modelica output - in principle, the flat model should be totally equivalent to the original one.

@perost can you please check why this happens? Thanks!

--- a/OMCompiler/Compiler/Template/CodegenUtil.tpl
+++ b/OMCompiler/Compiler/Template/CodegenUtil.tpl
@@ -90,9 +90,14 @@ import ExpressionDumpTpl.*;
 end symbolName;
 
 template replaceDotAndUnderscore(String str)
- "Replace _ with __ and dot in identifiers with _"
+ "Replace _ with __ and dot in identifiers with _.
+  For quoted identifiers (those starting with a single quote) the dots are
+  part of the name itself (e.g. 'Modelica.Media.X') and must be sanitized by
+  unquoteIdentifier (-> _2E), not treated as path separators (-> _5F_5F).
+  Otherwise the generated name diverges from the record type name built by
+  AbsynUtil.pathStringUnquoteReplaceDot and the C code fails to compile. See #13009."
 ::=
-  let str_dots = System.stringReplace(str,".", "_")
+  let str_dots = if stringEq(System.substring(str,1,1), "'") then str else System.stringReplace(str,".", "_")
   let str_underscores = System.stringReplace(str_dots, "_", "__")
   System.unquoteIdentifier(str_underscores)
 end replaceDotAndUnderscore;

--- a/testsuite/simulation/modelica/records/Makefile
+++ b/testsuite/simulation/modelica/records/Makefile
@@ -11,6 +11,7 @@ RecordConstructor1.mos \
 TestComplexSum1.mos \
 TestComplexSum2.mos \
 Ticket5134.mos \
+Ticket13009.mos \
 Ticket14485.mos \
 
 # test that currently fail. Move up when fixed. 

--- a/testsuite/simulation/modelica/records/Ticket13009.mos
+++ b/testsuite/simulation/modelica/records/Ticket13009.mos
@@ -1,0 +1,39 @@
+// name:     Ticket13009.mos
+// keywords: record quoted identifier dot codegen
+// status:   correct
+// teardown_command: rm -rf P.M* _P.M* output.log
+
+// Quoted identifiers that contain a dot (e.g. the record name 'fluid.R')
+// must be mangled consistently between the generated record typedef and the
+// places where that type is used. Previously the use sites replaced the dot
+// inside the quoted identifier with __ (-> _5F_5F) while the typedef kept it
+// as part of the name (-> _2E), so the C code failed to compile. See #13009.
+
+loadString("
+package 'P'
+  record 'fluid.R'
+    Real x;
+  end 'fluid.R';
+
+  model 'M'
+    'fluid.R' r;
+  equation
+    r = 'fluid.R'(time);
+  end 'M';
+end 'P';
+"); getErrorString();
+
+simulate('P'.'M'); getErrorString();
+
+// Result:
+// true
+// ""
+// record SimulationResult
+//     resultFile = "_omcQ_27P_27._omcQ_27M_27_res.mat",
+//     simulationOptions = "startTime = 0.0, stopTime = 1.0, numberOfIntervals = 500, tolerance = 1e-6, method = 'dassl', fileNamePrefix = '_omcQ_27P_27._omcQ_27M_27', options = '', outputFormat = 'mat', variableFilter = '.*', cflags = '', simflags = ''",
+//     messages = "LOG_SUCCESS       | info    | The initialization finished successfully without homotopy method.
+// LOG_SUCCESS       | info    | The simulation finished successfully.
+// "
+// end SimulationResult;
+// ""
+// endResult

